apps/vetrina/src/Vetrina/Main.purs

```diff
@@ -231,14 +235,27 @@ render self =
             , children: [ DOM.text "OK" ]
             }
         ]
-    PurchaseUnexpectedError -> DOM.text "SOMETHING WENT HORRIBLY WRONG SERVER SIDE"
-
-vetrinaContainer :: Array JSX -> JSX
-vetrinaContainer children =
-  DOM.div
-    { className: "vetrina--container"
-    , children
-    }
+    PurchaseUnexpectedError -> vetrinaContainer self $ Array.singleton $
+      Purchase.Error.error
+        { onRetry: onRetry
+        }
+  where
+    onRetry = do
+      Aff.launchAff_ $ tryMagicLogin self
+      self.setState _ { purchaseState = NewPurchase }
+
+vetrinaContainer :: Self -> Array JSX -> JSX
+vetrinaContainer self@{ state: { purchaseState } } children =
+  let errorClassString = "vetrina--purchase-error"
+      errorClass       = case purchaseState of
+                           PurchaseFailed          -> " " <> errorClassString
```

Comment:
Just a personal opinion, but somehow just adding a space to the end of `className: "vetrina--container " ...` seems cleaner to me, rather than adding the space here 😄